pcsx2-qt/MainWindow.cpp

```diff
@@ -1291,30 +1295,37 @@ void MainWindow::onInputRecPlayActionTriggered()
 	{
 		fileNames = dialog.selectedFiles();
 	}
+	else
+	{
+		if (!wasPaused)
+		{
+			VMManager::SetPaused(false);
+			return;
+		}
+	}
 
 	if (fileNames.length() > 0)
 	{
-		if (g_InputRecording.IsActive())
+		if (g_InputRecording.isActive())
 		{
-			g_InputRecording.Stop();
+			g_InputRecording.stop();
+			m_ui.actionInputRecStop->setEnabled(false);
 		}
-		if (g_InputRecording.Play(fileNames.first().toStdString()))
+		if (g_InputRecording.play(fileNames.first().toStdString()))
 		{
+			m_ui.actionInputRecStop->setEnabled(true);
 			return;
 		}
 	}
-
-	if (!wasPaused)
-	{
-		g_InputRecordingControls.Resume();
-	}
 }
 
 void MainWindow::onInputRecStopActionTriggered()
 {
-	if (g_InputRecording.IsActive())
+	if (g_InputRecording.isActive())
 	{
-		g_InputRecording.Stop();
+		g_InputRecording.stop();
```

Comment:
Should this execute on the EE/CPU thread too? (`Host::RunOnCPUThread()`)

Reply:
What is the accepted pattern for getting a value out of the CPU thread to inform a UI action?  For example below this line:
```cpp
if (g_InputRecording.play(fileNames.first().toStdString()))
{
	m_ui.actionInputRecStop->setEnabled(true);
	return;
}
```
I assume I should also be running `play` on the CPU thread, but how am I supposed to get the bool result out to perform further UI actions?  I assume `RunOnCPUThread` is not blocking, as well as it currently returns `void` so doing something like the following wouldn't help anyway.

```cpp
Host::RunOnCPUThread([&]() {
	if (g_InputRecording.play(fileNames.first().toStdString()))
	{
		m_ui.actionInputRecStop->setEnabled(true);
		return;
	}
});
```